### PR TITLE
Add SqlWalker to sort tuples with NULL values on Postgresql database

### PR DESCRIPTION
**Pull request by jerome-fix**

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
Could you rebase your PR? Wrong tests on Travis.

**Comment by Soullivaneuh** [comment]
Thanks. Please give some explanation of your PR with some samples then contributors can review it more easily.

**Comment by jerome-fix** [comment]
Hi,

I added some explanations with code.

You can see more on this link : http://troels.arvin.dk/db/rdbms/#select-order_by

I also add MySQL support. You can put NULL elements in the end.

You can also see a particular usage in SonataClassification Bundle : 
https://github.com/sonata-project/SonataClassificationBundle/pull/108

With PostGresql it is always the most little and not null parent who is returned 
as first one… and so an exception is thrown : « The first category must be the root ».

**Comment by jerome-fix** [comment]
ping @Soullivaneuh

**Comment by Soullivaneuh** [comment]
I just made a review. I'm not the best person to review and accept (or not) this PR. :-)

File: src/Query/SortableNullsWalker.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ *				"p.firstname" => Webges\DoctrineExtensions\Query\SortableNullsWalker::NULLS_FIRST,
+ *				"p.lastname"  => Webges\DoctrineExtensions\Query\SortableNullsWalker::NULLS_LAST,
+ *				"p.basedOnPerson.id" => Webges\DoctrineExtensions\Query\SortableNullsWalker::NULLS_LAST
+ *			));
+ */
+class SortableNullsWalker extends Query\SqlWalker
+{
+    const NULLS_FIRST = 'NULLS FIRST';
+
+    const NULLS_LAST = 'NULLS LAST';
```
Comment (Soullivaneuh) [inline src/Query/SortableNullsWalker.php]: Please group constant and align equal signs.

File: tests/Entities/BlogPost.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?php
+
```
Comment (Soullivaneuh) [inline tests/Entities/BlogPost.php]: Missing Sonata header.

File: tests/Mocks/DatabasePlatformMock.php [hunk 3]
```diff
@@ -0,0 +1,148 @@
+<?php
+
```
Comment (Soullivaneuh) [inline tests/Mocks/DatabasePlatformMock.php]: Missing Sonata header.

File: tests/Mocks/DatabasePlatformMock.php [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getClobTypeDeclarationSQL(array $field)
+    {
+    }
+
+    /* MOCK API */
```
Comment (Soullivaneuh) [inline tests/Mocks/DatabasePlatformMock.php]: Why this comment?

File: tests/Mocks/DriverConnectionMock.php [hunk 5]
```diff
@@ -0,0 +1,80 @@
+<?php
+
```
Comment (Soullivaneuh) [inline tests/Mocks/DriverConnectionMock.php]: Missing Sonata header.

File: tests/Mocks/DriverMock.php [hunk 6]
```diff
@@ -0,0 +1,85 @@
+<?php
+
```
Comment (Soullivaneuh) [inline tests/Mocks/DriverMock.php]: Missing Sonata header.

File: tests/Mocks/DriverMock.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+namespace Sonata\Doctrine\tests\Mocks;
+
+/**
+ * Mock class for Driver.
+ */
+class DriverMock implements \Doctrine\DBAL\Driver
```
Comment (Soullivaneuh) [inline tests/Mocks/DriverMock.php]: Import class and don't use FQN.

File: tests/Mocks/DriverMock.php [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+{
+    /**
+     * @var \Doctrine\DBAL\Platforms\AbstractPlatform|null
```
Comment (Soullivaneuh) [inline tests/Mocks/DriverMock.php]: Import class and don't use FQN.

File: tests/Mocks/DriverMock.php [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+     */
+    private $_platformMock;
+
+    /**
+     * @var \Doctrine\DBAL\Schema\AbstractSchemaManager|null
```
Comment (Soullivaneuh) [inline tests/Mocks/DriverMock.php]: Import class and don't use FQN.

**Comment by Soullivaneuh** [comment]
Some of my code comments must be applied on the rest of the code.

File: src/Query/SortableNullsWalker.php (hunk 1 above)
Comment (jerome-fix) [inline src/Query/SortableNullsWalker.php]: Group constant : Ok.

**Comment by jerome-fix** [comment]
Thank you for your code comments !
I tried to apply all your recommendations and I just push the new code.

**Comment by Soullivaneuh** [comment]
Thanks. I let @rande decide for this PR.

**Comment by jerome-fix** [comment]
Could this PR be merged? Thanks.

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**